WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift

```diff
@@ -171,6 +184,12 @@ private extension ApplicationPasswordAuthorizationWebViewController {
         }
     }
 
+    @objc private func popBothControllers() {
+        // Pop back two view controllers to remove both error and web view
+        navigationController?.popViewController(animated: false)
+        navigationController?.popViewController(animated: true)
```

Comment:
I think this is hacky and prone to error. How about looking for the index of `ApplicationPasswordTutorialViewController` in the navigation stack, and pop to the index before that? 

A better option would be to send the view controller before the application password flow to this view controller via `presentApplicationPasswordWebView` in `AuthenticationManager`.

Reply:
I agree with you and with your second suggestion, so I applied the new changes in this commit: 9b480ba48fc8c2f0c04aac77c43c512fb103c82e